R/hiper_glm.R

```diff
@@ -1,17 +1,47 @@
 #' @export
-hiper_glm <- function(design, outcome, model="linear") {
+hiper_glm <- function(design, outcome, model="linear", option = list()) {
 
   supported_model <- c("linear", "logit")
 
   if (!(model %in% supported_model)) {
     stop(sprintf("The model %s is not supported.", model))
   }
 
-  warning("`hiper_glm` is yet to be implemented.")
+  coef_estimate <- NULL
+
+  if (!is.null(option[["mle_solver"]])) {
+
+    if (option[["mle_solver"]] == "BFGS") {
```

Comment:
If you're going to let people specify BFGS and the default is pseudo inv, I'd recommend also letting them specify pseudo inv explicitely. It's a bit weird if the default is not an explicitely specifiable option.